### ISSUE TITLE
High number of get_posts queries

### ISSUE THREAD

**Issue by claytoncollie** [wpmetabox/mb-relationships#6]
I have 17 custom post types. And although not all will be loaded on a production site, each one relates to one main CPT. I normally have 3-6 cpts on a production site. Person is the main CPT and then all other cpts like presentation, publication, offices all relate to person. On each page reload in the admin, I am calling get_posts a ton, I think, because the metabox is querying each post type to populate the select fields.  

Is there anything we can do to bundle these queries or even use WP_Query so caching can be utilized? 

Should I `suppress_filters => false` ?  

My main goal is reduce the number of queries meaning that I am open to ideas.

![image](https://user-images.githubusercontent.com/4236538/43098567-0f27329e-8e85-11e8-9b04-021c3aeb63ed.png)

![image](https://user-images.githubusercontent.com/4236538/43098595-28780322-8e85-11e8-8efd-2471090afacd.png)

![image](https://user-images.githubusercontent.com/4236538/43098748-a5ef6796-8e85-11e8-801d-eef018ae577b.png)

**Comment by rilwis** [comment]
Hi @claytoncollie , please update [Meta Box](https://metabox.io) to version 4.15.1, the latest one. It has a fix for this problem. For more info, please see this [blog post](https://metabox.io/meta-box-weekly-updates-july-2018/).

**Comment by claytoncollie** [comment]
The screenshots I sent you were from 4.15.0.  I updated to 4.15.1 with no change in the number of queriees.  From your blog post and from the meta-box changelog, it looks like you are addressing clonable fields.  These queries are coming from the generation of the metabox themeselves, not a cloneable field. Right? or are you calling this a clonable field when each select field is being duplicated?

![image](https://user-images.githubusercontent.com/4236538/43142422-f9cace76-8f1d-11e8-9eba-185096a2e57f.png)

**Comment by rilwis** [comment]
Hi @claytoncollie ,

This is actually a cloneable `post` field. Previously, each value makes a query to get all available posts. The version 4.15 removes that behaviour, making sure all values make a single query.

I'm checking it more. Looks like there are some extra queries are made.

I also see so many queries are made by `update_meta_cache`, but can't see which component it is. Can you create a symlink to `db.php` of Query Monitor to show more info? [Here](https://github.com/johnbillion/query-monitor/wiki/db.php-Symlink) is the guide.

**Comment by claytoncollie** [comment]
I create a symlink properly but QM still is not recognizing it.  

I saw that update_post_meta was making a ton of queries and I thought it was because I had left over options in the wp_options table. The site I am showing you is my sandbox so I've installed over 60 plugins at some point on this DB. Obviously not all of them are active.

I actually had over 500 users registered on the site. That is what was slowing it down with update_post_meta. I removed almost all of the users and now the number of queries is way down.

**Comment by rilwis** [comment]
Hi @claytoncollie ,

I made an improvement for `user` field that might remove the number of queries.

https://github.com/wpmetabox/meta-box/commit/239d67f57b25f4ac568e373e31e9a4025f364bdc

Can you try the latest version on Github and let me know how it goes?

**Comment by claytoncollie** [comment]
Hi @rilwis ,

Thanks for looking into this. I had to delete all of those users in the meantime so my test will not be the same as before. My test site was too slow for me to work efficiently.

I added the snippet of code to my files and it saved me the exact number of queries as I have users. I went from 257 queries to 253 queries. And the field still works as expected.  I'd say this is a good commit. Feel free to close this issue.

Maybe to test on your end, create 400 users on an install and test a before and after.
